### PR TITLE
GH-30 initial integration of golangci-lint

### PR DESCRIPTION
**Pull request by hectorgabucio**
#### Summary

Integrates GolangCI-Lint in mattermost jenkins plugin. 
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-plugin-jenkins/issues/30

### CODE REVIEW

File: server/api.go [hunk 1]
```diff
@@ -86,13 +88,15 @@ func (p *Plugin) handleJobCreation(w http.ResponseWriter, r *http.Request) {
 	for k, v := range request.Submission {
 		jobInputs[k] = v.(string)
 	}
-	p.sendJobCreateRequest(userID, request.ChannelId, jobInputs)
+	if err := p.sendJobCreateRequest(userID, request.ChannelId, jobInputs); err != nil {
+		log.Println("Error sending job creation request", err)
```
Comment (hanzei) [inline server/api.go:92]: Please use `p.API.LogWarn` instead

File: server/plugin.go [hunk 2]
```diff
@@ -72,7 +73,10 @@ func (p *Plugin) OnActivate() error {
 		return errors.Wrap(appErr, "failed to set profile image")
 	}
 
-	p.API.RegisterCommand(getCommand())
+	if cmdErr := p.API.RegisterCommand(getCommand()); cmdErr != nil {
+		log.Println("Error registering Jenkins custom command", cmdErr)
```
Comment (hanzei) [inline server/plugin.go:77]: Same here

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the contribution :+1: 

Plugins use `p.API.Log*` instead of `log`. See https://developers.mattermost.com/extend/plugins/server/reference/#API.LogWarn for the docs. Would you please change that?

File: server/plugin.go [hunk 3]
```diff
@@ -72,7 +72,10 @@ func (p *Plugin) OnActivate() error {
 		return errors.Wrap(appErr, "failed to set profile image")
 	}
 
-	p.API.RegisterCommand(getCommand())
+	if cmdErr := p.API.RegisterCommand(getCommand()); cmdErr != nil {
+		p.API.LogWarn("Error registering Jenkins custom command", "err", cmdErr)
+	}
```
Comment (hanzei) [inline server/plugin.go:77]: Nit pick: It's idiomatic to use just `err` as variable name if there is no naming conflict.

**Review by hanzei — APPROVED** [review]
One nit pick, the rest looks good :+1:

**Review by hanzei — APPROVED** [review]
Great work :tada:

**Comment by hanzei** [comment]
@hectorgabucio Could you please resolve the merge conflict?

**Review by mickmister — APPROVED** [review]
Awesome work @hectorgabucio!

**Comment by hectorgabucio** [comment]
@hanzei the linting failed https://app.circleci.com/pipelines/github/mattermost/mattermost-plugin-jenkins/75/workflows/be714750-99de-48fe-927c-f340a226f6ca/jobs/246 maybe we use a newer golang ci version in CI?

**Comment by hectorgabucio** [comment]
@hanzei @mickmister conflicts resolved and CI passed

**Comment by hanzei** [comment]
Awesome, thanks @hectorgabucio :+1: 

@DHaussermann There are no function changes in this PR. I've successfully build and deployed the plugin locally. Any additional tests you want to make?

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Built from this branch - no lint issues with full `make`
- Enabled the plugin and tested that slash commands seem functional
- I will this during end to end testing before the next release for a thorough regression test
LGTM!
Thanks @hectorgabucio for this enhancement!

**Comment by jasonblais** [comment]
cc @mickmister @hanzei please help with merging if `Automerge` label is unsuccessful.

**Pull request merged**